Source/WebCore/loader/soup/ResourceLoaderSoup.cpp

```diff
@@ -33,12 +33,45 @@
 #include "ResourceError.h"
 #include "SharedBuffer.h"
 #include <gio/gio.h>
+#include <optional>
+#include <wtf/SortedArrayMap.h>
 #include <wtf/glib/GRefPtr.h>
 #include <wtf/glib/GUniquePtr.h>
 #include <wtf/glib/RunLoopSourcePriority.h>
+#include <wtf/text/ASCIILiteral.h>
+#include <wtf/text/WTFString.h>
 
 namespace WebCore {
 
+static std::optional<String> contentTypeLookUpForKnownResource(const char* filename)
```

Comment:
IMO the name of the function does not match what it does, which is plainly getting the MIME type from the file extension, from a fixed set of built-in mappings. How about something like “contentTypeForKnownExtension”?

```suggestion
static std::optional<String> contentTypeForKnownExtension(const char* filename)
```